v2/pkg/scan/scan_unix.go

```diff
@@ -182,8 +184,92 @@ func TransportReadWorkerPCAPUnix(s *Scanner) {
 		return
 	}
 
+	transportReaderCallback := func(tcp layers.TCP, udp layers.UDP, ip, srcIP4, srcIP6 string) {
+		// We consider only incoming packets
+		tcpPortMatches := tcp.DstPort == layers.TCPPort(s.SourcePort)
+		udpPortMatches := udp.DstPort == layers.UDPPort(s.SourcePort)
+		sourcePortMatches := tcpPortMatches || udpPortMatches
+		switch {
+		case !sourcePortMatches:
+			gologger.Debug().Msgf("Discarding Transport packet from non target ips: ip4=%s ip6=%s tcp_dport=%d udp_dport=%d\n", srcIP4, srcIP6, tcp.DstPort, udp.DstPort)
+
+		case s.Phase.Is(HostDiscovery):
+			proto := protocol.TCP
+			if udpPortMatches {
+				proto = protocol.UDP
+			}
+			s.hostDiscoveryChan <- &PkgResult{ip: ip, port: &port.Port{Port: int(tcp.SrcPort), Protocol: proto}}
+		case tcpPortMatches && tcp.SYN && tcp.ACK:
+			s.tcpChan <- &PkgResult{ip: ip, port: &port.Port{Port: int(tcp.SrcPort), Protocol: protocol.TCP}}
+		case udpPortMatches && udp.Length > 0: // needs a better matching of udp payloads
+			s.udpChan <- &PkgResult{ip: ip, port: &port.Port{Port: int(udp.SrcPort), Protocol: protocol.UDP}}
+		}
+	}
+
+	// In case of OSX, when we decode the data from 'loO' interface
+	// always get [Ethernet] layer only.
+	// with the help of data received from packetSource.Packets() we can
+	// extract the high level layers like [IPv4, IPv6, TCP, UDP]
+	macLoopBackScanCaseCallback := func(handler *pcap.Handle, wg *sync.WaitGroup) {
+		defer wg.Done()
+		packetSource := gopacket.NewPacketSource(handler, handler.LinkType())
+		for packet := range packetSource.Packets() {
```

Comment:
Wouldn't this drain packets from the underlying cgo pcap buffer which is also used for sequential read at https://github.com/projectdiscovery/naabu/blob/49eb0515dbe30d2a3edad1b93e9aaeb4549e6efa/v2/pkg/scan/scan_unix.go#L300 ?